test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -611,16 +612,17 @@ def __get_param_dictionary(self):
         for pkey in paramkeys:
             if pkey in ['servers', 'moms', 'comms', 'client']:
                 if not tparam_contents[pkey]:
-                    hostname = socket.gethostname()
-                    tparam_contents[pkey] = set([hostname.split('.', 1)[0]])
+                    tparam_contents[pkey] = set([shortname])
                 else:
                     tparam_contents[pkey] = set(tparam_contents[pkey])
         if not ((tparam_contents['servers']) & (tparam_contents['moms'])):
             tparam_dic['no_mom_on_server'] = True
         if not ((tparam_contents['servers']) & (tparam_contents['comms'])):
             tparam_dic['no_comm_on_server'] = True
-        if ((tparam_contents['moms']) & (tparam_contents['comms'])):
-            tparam_dic['no_comm_on_mom'] = False
+        commtrack = ((tparam_contents['moms']) & (tparam_contents['comms']))
+        if commtrack:
+            if not ((len(commtrack) == 1) and (shortname in commtrack)):
+                tparam_dic['no_comm_on_mom'] = False
```

Comment:
As we discussed, we don't need to calculate these flags value, we just have to use it in if else condition.